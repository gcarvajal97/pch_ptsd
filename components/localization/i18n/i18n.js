export const fallback = "en";
 
export const supportedLocales = {
    en: {
        name: "English",
        translationFileLoader: () => require('../lang/en.json'),
        // en is default locale in Moment
        momentLocaleLoader: () => Promise.resolve(),
    },
    es: {
        name: "EspaÃ±ol",
        translationFileLoader: () => require('../lang/es.json'),
        momentLocaleLoader: () => import('moment/locale/es'),
    },
};
 
export const defaultNamespace = "LearnMore";
 
export const namespaces = [
    "LearnMore"
];