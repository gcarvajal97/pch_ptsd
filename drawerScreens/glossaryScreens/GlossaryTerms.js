import React from "react";
import { StyleSheet, Text, View } from "react-native";

function GlossaryTerms(content) {
    if (content && content.length) {
        return (
            <View>
                {content.map((item, index) => {
                    return (
                        <View style={styles.contentContainer} key={index}>
                            <Text style={styles.term}>{item.term}</Text>
                            <Text style={styles.definition}>
                                {item.definition}
                            </Text>
                        </View>
                    );
                })}
            </View>
        );
    } else {
        return (
            <View>
                <Text>Oops, glossary term content is invalid!</Text>
            </View>
        );
    }
}

export default GlossaryTerms;

const styles = StyleSheet.create({
    definition: {
        fontSize: 14,
        marginVertical: 2
    },
    term: {
        fontSize: 14,
        fontWeight: "bold",
        marginVertical: 2
    },
    contentContainer: {
        padding: 20,
        backgroundColor: "white"
    }
});
