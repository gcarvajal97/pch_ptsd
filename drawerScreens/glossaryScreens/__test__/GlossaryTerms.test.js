import React from 'react';
import GlossaryTerms from '../GlossaryTerms';
import renderer from 'react-test-renderer';
import Enzyme, { shallow, configure } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import i18n from "i18n-js";

configure({ adapter: new Adapter() });
jest.mock('');

describe("GlossaryTerms", () => {
    
    let wrapper;

    beforeEach(() => {
        i18n.local = "en";
    });

    it('renders correctly', () => {
        const tree = renderer.create(<GlossaryTerms />).toJSON();
        expect(tree).toMatchSnapshot();
    });

    it('renders correctly in Spanish', () => {
        i18n.locale = 'es'
        const tree = renderer.create(<GlossaryTerms />).toJSON();
        expect(tree).toMatchSnapshot();
    });

});
