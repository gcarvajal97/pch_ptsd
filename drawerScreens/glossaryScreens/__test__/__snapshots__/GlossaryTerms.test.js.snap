// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GlossaryTerms should render a glossary term 1`] = `
<View>
  <View
    style={
      Object {
        "backgroundColor": "#FFFFFF",
        "padding": 18,
        "paddingVertical": 10,
      }
    }
  >
    <Text
      style={
        Object {
          "fontFamily": "avenir-heavy",
          "fontSize": 14,
        }
      }
    >
      a term
    </Text>
    <Text
      style={
        Object {
          "fontSize": 14,
        }
      }
    >
      a definition
    </Text>
  </View>
</View>
`;

exports[`GlossaryTerms should render an error message if glossary terms are not defined 1`] = `
<View>
  <Text>
    Oops, glossary term content is invalid!
  </Text>
</View>
`;

exports[`GlossaryTerms should render an error message if no glossary terms 1`] = `
<View>
  <Text>
    Oops, glossary term content is invalid!
  </Text>
</View>
`;

exports[`GlossaryTerms should render glossary terms 1`] = `
<View>
  <View
    style={
      Object {
        "backgroundColor": "#FFFFFF",
        "padding": 18,
        "paddingVertical": 10,
      }
    }
  >
    <Text
      style={
        Object {
          "fontFamily": "avenir-heavy",
          "fontSize": 14,
        }
      }
    >
      a term
    </Text>
    <Text
      style={
        Object {
          "fontSize": 14,
        }
      }
    >
      a definition
    </Text>
  </View>
  <View
    style={
      Object {
        "backgroundColor": "#FFFFFF",
        "padding": 18,
        "paddingVertical": 10,
      }
    }
  >
    <Text
      style={
        Object {
          "fontFamily": "avenir-heavy",
          "fontSize": 14,
        }
      }
    >
      another term
    </Text>
    <Text
      style={
        Object {
          "fontSize": 14,
        }
      }
    >
      another definition
    </Text>
  </View>
</View>
`;
